ipfshttpclient/filescanner.py

```diff
@@ -21,32 +21,22 @@
 	re_pattern_t = re_pattern_type = type(re.compile(""))
 
 
-if hasattr(enum, "auto"):  #PY36+
-	enum_auto = enum.auto
-elif not ty.TYPE_CHECKING:  #PY35
-	_counter = 0
-	
-	def enum_auto() -> int:
-		global _counter
-		_counter += 1
-		return _counter
-
-
-O_DIRECTORY = getattr(os, "O_DIRECTORY", 0)  # type: int
+# Windows does not have os.O_DIRECTORY
+O_DIRECTORY: int = getattr(os, "O_DIRECTORY", 0)
 
 
-HAVE_FWALK       = hasattr(os, "fwalk")  # type: bool
-HAVE_FWALK_BYTES = HAVE_FWALK and sys.version_info >= (3, 7)  # type: bool
+# Mac does not have os.fwalk even thru Python 3.9
```

Comment:
Curious, I always thought that macOS would support this! Are you sure it doesn't (since the `openat` system call required for this is mandatory since POSIX.1-2008?)

Regarding the comment, please change it to `macOS/Windows` instead of just `Mac`.

Reply:
The mac build failed stating `os.fwalk` wasn't available.

Reply:
True! True!

I never noticed it babbling about lack of fwalk support in the CI logs on macOS:

> SKIPPED [1] test/functional/test_files.py:262: No point in disabling os.fwalk if it isn't actually supported
> SKIPPED [1] test/functional/test_files.py:271: Passing directory as file descriptor requires os.fwalk
> SKIPPED [1] test/functional/test_files.py:287: No point in disabling os.fwalk if it isn't actually supported
> SKIPPED [1] test/functional/test_files.py:309: Passing directory as file descriptor requires os.fwalk
> SKIPPED [1] test/functional/test_files.py:326: No point in disabling os.fwalk if it isn't actually supported

:astonished: 

Looking around the internet this may actually be a bug introduced in Python 3.6 and never fixed…